src/routes/manage-disciplines/ManageDisciplinesScreen.tsx

```diff
@@ -0,0 +1,119 @@
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { ReactElement, useCallback } from 'react'
+import { Pressable } from 'react-native'
+import { Subheading } from 'react-native-paper'
+import styled from 'styled-components/native'
+
+import { CloseIconRed } from '../../../assets/images'
+import AddElement from '../../components/AddElement'
+import HorizontalLine from '../../components/HorizontalLine'
+import ListItem from '../../components/ListItem'
+import ServerResponseHandler from '../../components/ServerResponseHandler'
+import { Heading } from '../../components/text/Heading'
+import { Discipline } from '../../constants/endpoints'
+import labels from '../../constants/labels.json'
+import useReadCustomDisciplines from '../../hooks/useReadCustomDisciplines'
+import useReadSelectedProfessions from '../../hooks/useReadSelectedProfessions'
+import { RoutesParams } from '../../navigation/NavigationTypes'
+import { removeSelectedProfession } from '../../services/AsyncStorage'
+import { reportError } from '../../services/sentry'
+import CustomDisciplineItem from './components/CustomDisciplineItem'
+
+const Root = styled.ScrollView`
+  display: flex;
+  background-color: ${props => props.theme.colors.background};
+  padding: ${props => props.theme.spacings.md};
+`
+
+const SectionHeading = styled(Subheading)`
+  padding-top: ${props => props.theme.spacings.xl};
+`
+
+const Padding = styled.View`
+  padding-bottom: ${props => props.theme.spacings.xxl};
+`
+
+interface Props {
+  navigation: StackNavigationProp<RoutesParams, 'ManageDisciplines'>
+}
+
+const ManageDisciplinesScreen = ({ navigation }: Props): ReactElement => {
+  const {
+    data: selectedProfessions,
+    loading,
+    error,
+    refresh: refreshSelectedProfessions
+  } = useReadSelectedProfessions()
+  const { data: customDisciplines, refresh: refreshCustomDisciplines } = useReadCustomDisciplines()
+
+  const refresh = useCallback(() => {
+    refreshCustomDisciplines()
+    refreshSelectedProfessions()
+  }, [refreshCustomDisciplines, refreshSelectedProfessions])
+
+  React.useEffect(() => {
+    refresh()
+    const willFocusSubscription = navigation.addListener('focus', () => {
+      refresh()
+    })
+    return willFocusSubscription
+  }, [navigation, refresh])
```

Comment:
Any reason why you are not using the focus effect from reactnavigation?
Minor: Unnecessary wrapping of refresh in the listener and no need for a const, can be returned directly.

Reply:
not sure, why I did this. Tried with FocusEffect and seems to work fine also.